src/core/public/chrome/chrome_service.tsx

```diff
@@ -94,11 +94,36 @@ export class ChromeService {
     injectedMetadata,
     notifications,
   }: StartDeps): Promise<InternalChromeStart> {
-    const FORCE_HIDDEN = isEmbedParamInHash();
+    /**
+     * These observables allow consumers to toggle the chrome visibility via either:
+     *   1. Using setIsVisible() to trigger the next chromeHidden$
+     *   2. Setting `chromeless` when registering an application, which will
+     *      reset the visibility whenever the next application is mounted
+     *   3. Having embed=true in the query string
+     */
+    const chromeHidden$ = new BehaviorSubject(false);
+    const forceHidden$ = of(isEmbedParamInHash());
+    const appHidden$ = merge(
+      // Default the app being hidden to the same value as forceHidden$ in case the
+      // application service has not emitted an app ID yet, since we want to trigger
+      // combineLatest below regardless of having an application value yet.
+      forceHidden$,
+      application.currentAppId$.pipe(
+        map(
+          appId =>
+            !!appId &&
+            application.availableApps.has(appId) &&
+            !!application.availableApps.get(appId)!.chromeless
+        )
+      )
+    );
+    const isVisible$ = combineLatest(appHidden$, forceHidden$, chromeHidden$).pipe(
+      map(([appHidden, forceHidden, chromeHidden]) => !(appHidden || forceHidden || chromeHidden)),
```

Comment:
Now that the app definition takes precedence over setting `setIsVisible(true)` can we add a note to the doc on `ChromeStart.setIsVisible` that explains that case?